src/Tokens/Models/UserIdToken/UserIdToken.php

```diff
@@ -169,17 +169,24 @@ private function parseToken(): array
             throw new InvalidUserIdTokenException(sprintf('Unable to parse base64 string %s.', $payloadBase64String));
         }
 
-        $payload = json_decode($payloadJsonString);
+        $payload = json_decode($payloadJsonString, true);
 
         if ($payload === null) {
             throw new InvalidUserIdTokenException(sprintf('Unable to parse JSON string %s.', $payloadJsonString));
         }
 
-        return is_array($payload) ? $payload : (array)$payload;
+        if (!is_array($payload)) {
+            throw new InvalidUserIdTokenException(sprintf('Unable to parse array from %s', $payload));
```

Comment:
This should probably have `$payloadJsonString` in the error, rather than the `$payload` result.